File: business/business-oj/src/main/java/com/glowxq/oj/meta/enums/ImageBusinessType.java
package com.glowxq.oj.meta.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum ImageBusinessType implements BaseType {

    Carousel("Carousel", "轮播图"),
    Temp("Temp", "临时图片"),

    ;

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static ImageBusinessType matchCode(String code) {
        for (ImageBusinessType pushStatus : ImageBusinessType.values()) {
            if (pushStatus.getCode().equals(code)) {
                return pushStatus;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/oj/meta/ImageBusinessType.ts
import type { BaseEnum } from '@/enums/base';

// 图片业务类型枚举接口
interface ImageBusinessTypeEnum extends BaseEnum {
}

/**
 * 图片业务类型枚举
 * 用于标识不同的图片业务类型
 */
export const ImageBusinessType: { [key: string]: ImageBusinessTypeEnum } = {
  /**
   * 轮播图
   */
  CAROUSEL: {
    code: 'Carousel',
    name: '轮播图'
  },

  /**
   * 临时图片
   */
  TEMP: {
    code: 'Temp',
    name: '临时图片'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): ImageBusinessTypeEnum | null {
  return Object.values(ImageBusinessType).find(item => item.code === code) || null;
} 